wordclock/wificfg.c

```diff
@@ -704,8 +725,22 @@ static void handle_hw_cfg_post(int s, wificfg_method method,
 
             wificfg_form_url_decode(buf);
             printf("%s %s %s\n", __FUNCTION__, name, buf);
-            if (strcmp(name, "cl_hadrwaretype") == 0) {
+            if (strcmp(name, "hw_hadrwaretype") == 0) {
                 g_settings.hardwareType = atoi(buf);
+            } else if (strcmp(name, "hw_imperfections") == 0) {
+                if (strstr(buf, "CheckOn") != NULL) {
+                    g_settings.perfectImperfections = 1;
+                }
+            } else if (strcmp(name, "hw_hierbenik_url") == 0) {
+                strcpy(g_settings.hierbenikUrl, buf);
```

Comment:
strcpy is unsafe here, prone to error. Use for safety strncopy with with len: sizeoff(g_settings.hierbenikUrl) - 1.
See for example AnimationSetMessageText()